pkg/apis/core/types.go

```diff
@@ -5351,3 +5360,23 @@ type TopologySpreadConstraint struct {
 	// +optional
 	LabelSelector *metav1.LabelSelector
 }
+
+// PortStatus represents the condition of a service port
+type PortStatus struct {
+	// Port is the port number of the service port of which status is recorded here
+	Port int32
+	// Protocol is the protocol of the service port of which status is recorded here
+	Protocol Protocol
+	// Error is to record any problem with the service port
+	Error string
```

Comment:
Is the intention to populate this status block ONLY if there is an error?  Or to poplulate always and sometimes error is empty?  If the latter (which is what I expect) we should make error a pointer (optional)

Reply:
The intention is to use it in all cases. Modified the Error to string pointer.

Reply:
This question is back.  Let's be clear - if there's not an error on this port the struct is still populated in status...ports[] but the Error field is blank, right?

If so, the field itself should be optional and omitempty.